daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/GlobalKey.scala

```diff
@@ -25,12 +27,67 @@ final class GlobalKey private (
   // Ready for refactoring where packageId becomes optional (#14486)
   def packageId: Option[Ref.PackageId] = Some(templateId.packageId)
   def qualifiedName: Ref.QualifiedName = templateId.qualifiedName
+  def shared: SharedGlobalKey =
+    SharedGlobalKey(Some(templateId.packageId), templateId.qualifiedName, key, hash)
 
   override def hashCode(): Int = hash.hashCode()
 
   override def toString: String = s"GlobalKey($templateId, $key)"
 }
 
+// The target state for GlobalKey once all call sites support SharedGlobalKey (#14486)
```

Comment:
Can you create an issue about removing the `pakageId` in contract key and link it here 
```suggestion
// TODO: https://github.com/digital-asset/daml/issues/XXXX
//   The target state for GlobalKey once all call sites support SharedGlobalKey (#14486)
```